### ISSUE TITLE
Exception thrown when starting controller in initState (version 0.5.4)

### ISSUE THREAD

**Issue by kwellman** [funwithflutter/flutter_confetti#29]
Hello,

In version 0.5.3 there were no problems when starting the confetti controller in initState. In version 0.5.4, I get the following exception:

(Note: it works fine when I wrap my _confettiController.play() call within the WidgetsBinding.instance.addPostFrameCallback method... I've pasted my code below)

```
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown building ConfettiDialog(state: _ConfettiDialogState#b3e75):
flutter: dependOnInheritedWidgetOfExactType<MediaQuery>() or dependOnInheritedElement() was called before
flutter: _ConfettiWidgetState.initState() completed.
flutter: When an inherited widget changes, for example if the value of Theme.of() changes, its dependent
flutter: widgets are rebuilt. If the dependent widget's reference to the inherited widget is in a constructor
flutter: or an initState() method, then the rebuilt dependent widget will not reflect the changes in the
flutter: inherited widget.
flutter: Typically references to inherited widgets should occur in widget build() methods. Alternatively,
flutter: initialization based on inherited widgets can be placed in the didChangeDependencies method, which
flutter: is called after initState and whenever the dependencies change thereafter.
flutter:
flutter: The relevant error-causing widget was:
flutter:   ConfettiDialog 
lib/screens/test_screen.dart:54
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      StatefulElement.dependOnInheritedElement.<anonymous closure> 
package:flutter/…/widgets/framework.dart:4805
flutter: #1      StatefulElement.dependOnInheritedElement 
package:flutter/…/widgets/framework.dart:4848
flutter: #2      Element.dependOnInheritedWidgetOfExactType 
package:flutter/…/widgets/framework.dart:3885
flutter: #3      MediaQuery.of 
flutter: #4      _ConfettiWidgetState._getScreenSize 
package:confetti/src/confetti.dart:243
flutter: #5      _ConfettiWidgetState._setScreenSize 
package:confetti/src/confetti.dart:227
flutter: #6      _ConfettiWidgetState._startAnimation 
package:confetti/src/confetti.dart:211
flutter: #7      _ConfettiWidgetState._initAnimation 
package:confetti/src/confetti.dart:158
flutter: #8      _ConfettiWidgetState.initState 
package:confetti/src/confetti.dart:146
flutter: #9      StatefulElement._firstBuild 
package:flutter/…/widgets/framework.dart:4684
flutter: #10     ComponentElement.mount 
package:flutter/…/widgets/framework.dart:4520
flutter: ...     Normal element mounting (4 frames)
flutter: #14     Element.inflateWidget 
package:flutter/…/widgets/framework.dart:3490
flutter: #15     MultiChildRenderObjectElement.mount 
package:flutter/…/widgets/framework.dart:5991
flutter: #16     Element.inflateWidget 
package:flutter/…/widgets/framework.dart:3490
flutter: #17     Element.updateChild 
package:flutter/…/widgets/framework.dart:3255
flutter: #18     ComponentElement.performRebuild 
package:flutter/…/widgets/framework.dart:4571
flutter: #19     StatefulElement.performRebuild 
package:flutter/…/widgets/framework.dart:4719
flutter: #20     Element.rebuild 
package:flutter/…/widgets/framework.dart:4262
flutter: #21     BuildOwner.buildScope 
package:flutter/…/widgets/framework.dart:2667
flutter: #22     WidgetsBinding.drawFrame 
package:flutter/…/widgets/binding.dart:866
flutter: #23     RendererBinding._handlePersistentFrameCallback 
package:flutter/…/rendering/binding.dart:286
flutter: #24     SchedulerBinding._invokeFrameCallback 
package:flutter/…/scheduler/binding.dart:1117
flutter: #25     SchedulerBinding.handleDrawFrame 
package:flutter/…/scheduler/binding.dart:1056
flutter: #26     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> 
package:flutter/…/scheduler/binding.dart:865
flutter: (elided 4 frames from class _RawReceivePortImpl, class _Timer, and dart:async-patch)
flutter:
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
```

Here's my test code:

```
import 'dart:math' show pi;
import 'package:confetti/confetti.dart';
import 'package:flutter/material.dart';

class TestScreen extends StatefulWidget {
  @override
  _TestScreenState createState() => _TestScreenState();
}

class _TestScreenState extends State<TestScreen> {
  ConfettiController _confettiController;

  @override
  void initState() {
    super.initState();
    _confettiController = ConfettiController(duration: Duration(seconds: 3));
    // since confetti version 0.5.4 need to call play in post frame call back (bug?)
    // otherwise I get a "dependOnInheritedWidgetOfExactType<MediaQuery>() or dependOnInheritedElement() was called before _ConfettiWidgetState.initState() completed." exception.
    WidgetsBinding.instance.addPostFrameCallback((_) {
      _confettiController.play();      
    });
  }

  @override
  void dispose() {
    _confettiController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
      return Stack(
        alignment: AlignmentDirectional.center,
        children: [
          Align(
            alignment: Alignment(0, -1.1),
            child: ConfettiWidget(
              confettiController: _confettiController,
              blastDirection: pi / 2,
              blastDirectionality: BlastDirectionality.explosive,
              maxBlastForce: 14,
              minBlastForce: 4,
              emissionFrequency: 0.3,
              numberOfParticles: 2,
              gravity: 0.26,
            ),
          ),
          Text('Confetti!', style: Theme.of(context).textTheme.headline3.copyWith(color: Colors.white)),
        ],
    );
  }
}
```

**Comment by funwithflutter** [comment]
Thanks for the issue. Before v0.5.4 WidgetsBinding.instance.addPostFrameCallback was used to initialise the position of the widget. The latest change initialises the position on play, instead of initState, to fix an error with PageViews initialising the confetti at a wrong location when a new page is swiped in.

I can wrap the play call in WidgetsBinding.instance.addPostFrameCallback to resolve this issue.

**Comment by funwithflutter** [comment]
Fixed in v0.5.4+1 commit bfbc663b588cfc2c2ea24b5b4a56bcb1b6555c53
